SensorFusion_src/Support_Degee_Matrix.c

```diff
@@ -0,0 +1,33 @@
+/******************************************************
+* Author : Shyam Bhuptani 
+* Function : Support Degree Matrix 
+* Application : This function converts the input array[N] (the array of various 
+* temprature) into a SDM matrix of which is having degre N*N
+* Input : size of temprature array N as integer and temprature[N] of integers 
+* Output : N*N support degree matrix
+*/
+
+/*Including dependancies*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "../include/Support_Degree_Matrix.h"
+
+/*Function declaration which returns a 2D array */
+double ** Support_Degree_Matrix(int n, int temprature[n])
```

Comment:
Not sure why your argument is called "temperature". The sensors may measure quantities that are not temperature (e.g. pressure, humidity, etc.). I suggest that you find a better argument name. Also, typically, functions that pass in arrays with a fixed number of elements have the array as the first argument and the number of elements as the second argument:

returntype funcname(int array[], int n)